py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py

```diff
@@ -322,6 +334,31 @@ def _construct_trt_network_def(self) -> None:
             f"TRT INetwork construction elapsed time: {datetime.now() - run_module_start_time}"
         )
 
+    @staticmethod
+    def find_weight(
+        weight_name: str, np_map: dict[str, Any], sd: dict[str, Any]
+    ) -> str:
+        network_weight = np_map[weight_name]
+        network_weight = torch.from_numpy(np_map[weight_name]).cuda()
+        for sd_w_name, sd_weight in sd.items():
+            if TRTInterpreter.check_weight_equal(sd_weight, network_weight):
+                del sd[sd_w_name]
+                return sd_w_name
+        return ""
+
+    @staticmethod
+    def check_weight_equal(
+        sd_weight: torch.tensor, network_weight: Union[torch.Tensor, np.ndarray]
+    ) -> Any:
+        if not isinstance(network_weight, torch.Tensor):
+            network_weight = torch.from_numpy(network_weight).cuda()
```

Comment:
do we need .cuda() ? can we compare them on cpu ?

Reply:
We want to use cuda to accelerate the weight finding process, especially when checking whether two weights are equal